app/models/ems_event.rb

```diff
@@ -136,6 +137,18 @@ def self.process_host_in_event!(event, options = {})
     end
   end
 
+  def self.process_physical_storage_in_event!(event, options = {})
+    physical_storage_ems_ref = event[:physical_storage_ems_ref]
+    process_object_in_event!(PhysicalStorage, event, options)
+
+    if event[:physical_storage_id].nil? && physical_storage_ems_ref.present?
+      storage = PhysicalStorage.find_by(:ems_ref => physical_storage_ems_ref)
+      unless storage.nil?
+        event[:physical_storage_id] = storage.id
+      end
+    end
```

Comment:
@TonySchneider are you hitting this section of code?  I think it would only execute if ems_id were nil which shouldn't happen

Reply:
Yes, I have debugged it many times

Reply:
Okay can you tell which line of https://github.com/Autosde/manageiq/blob/8f8b88154663751d11bb058a95f5620dbc86ce5a/app/models/ems_event.rb#L93-L109 it fails on?  Is it that `event[:ems_id]` is `nil` or something else?

Reply:
I'm pretty sure that event[:ems_id] can't be nil due to this line:
https://github.com/Autosde/manageiq/blob/8f8b88154663751d11bb058a95f5620dbc86ce5a/app/models/ems_event.rb#L73